subprojects/test-runner/client/src/main/kotlin/com/avito/runner/scheduler/runner/TestRunnerImpl.kt

```diff
@@ -40,7 +40,16 @@ internal class TestRunnerImpl(
     private val logger = loggerFactory.create<TestRunner>()
 
     override suspend fun runTests(tests: List<TestCase>): Result<TestRunnerResult> {
-        return withTimeout(executionTimeout.toMillis()) {
+        return withTimeoutOrDefault(
+            timeMillis = executionTimeout.toMillis(),
+            default = Result.Failure(
```

Comment:
As I understand this code never be executed

Reply:
Test shows the behavior changes